app/controllers/admin/merchants_controller.rb

```diff
@@ -0,0 +1,18 @@
+class Admin::MerchantsController < Admin::BaseController
+  def index
+    @merchants = Merchant.all
+  end
+
+  def update
+    merchant = Merchant.find(params[:id])
+    if params[:activate] == "true"
+      merchant.activate
+      flash[:success] = "#{merchant.name} has been activated"
+      redirect_to "/admin/merchants"
+    else
+      merchant.deactivate
+      flash[:success] = "#{merchant.name} has been disabled"
+      redirect_to "/admin/merchants"
```

Comment:
Activates or deactivates the merchant depending on what button was presses. I think this can be cleaned up by using `toggle` to reverse the boolean value.